autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +121,79 @@ def get_dataset(self, segment='train'):
         sampler, _ = get_split_samplers(dataset, split_ratio=0.2)
         return SampledDataset(dataset, sampler)
 
-class TSVClassificationTask(AbstractGlueTask):
-    def __init__(self, *args, **kwargs): # passthrough arguments to TSVDataset
-        # (filename, field_separator=nlp.data.Splitter(','), num_discard_samples=1, field_indices=[2,1])
-        self.args = args
-        self.kwargs = kwargs
-        is_pair = False
-        class_labels = ['0', '1']
-        metric = CompositeEvalMetric()
-        metric.add(F1())
-        metric.add(Accuracy())
-        super(TSVClassificationTask, self).__init__(class_labels, metric, is_pair)
-        dataset = nlp.data.TSVDataset(*self.args, **self.kwargs)
-        # do the split
-        train_sampler, val_sampler = get_split_samplers(dataset, split_ratio=0.8)
-        self.trainset = SampledDataset(dataset, train_sampler)
-        self.valset = SampledDataset(dataset, val_sampler)
+class CustomTSVClassificationTask(AbstractGlueTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
```

Comment:
Add option for user to give a pandas DataFrame as input (train_data)

Reply:
For the initial tsv support and as normal practice in NLP, I would prefer just support the limited file type. It introduces less dependency and concept (e.g., dataframe) to the user, as it is a user interface.

Reply:
You are already loading it into a DataFrame internally. They don't need to use this. If you look at Tabular, train_data can be either str or DataFrame. I think this should act in the same way.

Reply:
This is a user interface. Regardless what's been implemented internally, I think it is more important to expose to NLP users what they are used to: dataset path, less dependency (as we already have many dependencies internally, do not pass the burden to the user further).